test/boost/sstable_3_x_test.cc

```diff
@@ -3382,9 +3382,8 @@ static void write_mut_and_validate_version(test_env& env, schema_ptr s, const ss
     lw_shared_ptr<memtable> mt = make_lw_shared<memtable>(s);
     mt->apply(mut);
 
-    tmpdir tmp = write_and_compare_sstables(env, s, mt, table_name, version);
+    tmpdir tmp = write_sstables(env, s, mt, version);
     auto written_sst = validate_read(env, s, tmp.path(), {mut}, version);
```

Comment:
We can disable the tests temporarily until we get rid of the converters in the pipeline. Then the flush should generate the same sstables as before.

Reply:
I'm not entirely sure that byte-by-byte comparison of range-tombstone-containing sstables is worth it in general (since the format does not specify a canonical range tombstone representation), but it definitely can be reinstated after we get rid of converters.  Completely disabling the affected tests because of this seems overkill (if that's what you are suggesting)

Reply:
We can disable the current version for now
and when the readers are native v2 restore the old version.

Reply:
Maybe we can add a canonicalizer for the tests?

Reply:
Those tests are about comparing generated sstables against a golden copy. By "canonicalizer" do you mean a writer which generates a mutation fragment stream which will yield the same sstables as the golden copy? In general, that's not possible because information is lost when v2 is produced, as v1->v2 is deoverlapping range tombstones. We don't know by looking at v2 in what way they overlapped, or whether they overlapped.

Comparing golden copy contents with tests input using flat_mutation_reader interface already works.